### DISCUSSION TITLE
Access token issue.

### DISCUSSION THREAD

**Discussion by marius92mc** [tweepy/tweepy#1760] [Questions] [answered]
I don't find any possibility to get access tokens anymore. Related to the updated tweepy, i didn't see mentioned the new way to get the access token.

When I have

``` python
verifier = request.GET.get('oauth_verifier')
try:
    oauth.get_access_token(verifier)
except tweepy.TweepError:
    print "Error, failed to get access token"
```

It always goes on exception part.

**Comment by marius92mc** [comment]
No solutions to this?

**Comment by martinarrieta** [comment]
What version are you using? 

Are you getting the key and secret from the request token correctly? 

Show us how you are setting the values to "oauth.set_request_token" before that and verify the content of the variables too.

Regards, 

Martin.

**Comment by marius92mc** [comment]
Yes, I get the keys correctly.
Which command should I use to see the version of tweepy?

https://gist.github.com/marius92mc/309ec53acbed0042c4e6

**Comment by martinarrieta** [comment]
To get the current version you can do :

➜  ~  ipython
In [1]: import tweepy
In [2]: tweepy.**version**
Out[2]: '3.2.0'

Let me know the version that you are using and we will be able to help you.
Regards, 

Martin

**Comment by marius92mc** [comment]
Hm, looks like it doesn't work to output the tweepy version.

In [1]: import tweepy
In [2]: tweepy.version
AttributeError                            Traceback (most recent call last)
<ipython-input-2-a337d306f2a9> in <module>()
----> 1 tweepy.version
AttributeError: 'module' object has no attribute 'version'

Any solution to this though? Also, martinarrieta, I've put a gist link to the snippets of a file where I've used tweepy with the initial mentioned issue.

**Comment by marius92mc** [comment]
Still no ideas for the issues presented by me above?

**Comment by marius92mc** [comment]
Does someone know a solution to the presented issue?

**Comment by MechaCoder** [comment]
do you have your Code Posted on github?
remove the try catch statement and run the code and copy me the error?

**Comment by gtrdotmcs** [comment]
Here You can set,
`oauth.request_token = {'oauth_token': session['request_token'] ,'oauth_token_secret': session['verifier'] }`
 `get_token = oauth.get_access_token(session['verifier'])`
`print(oauth.access_token)`
`print(oauth.access_token_secret)`

if these are present then you are all set
this because oauth.request_token is not set properly.

**Comment by Harmon758** [comment, marked as answer]
I've rewritten the authentication documentation with https://github.com/tweepy/tweepy/commit/f9a722bae8cce368b9f8fd447c418e1034c32178, which should hopefully answer this question.

You can refer to the [3-legged OAuth](https://tweepy.readthedocs.io/en/latest/authentication.html#legged-oauth) section.

You can use [`pip show`](https://pip.pypa.io/en/stable/cli/pip_show/) or `tweepy.__version__` to check your version of Tweepy.
